### PR TITLE
fix(kuma-cp): don't add namespace labels when resource was synced from universal zone

### PR DESCRIPTION
**Pull request by Automaat**
Fixes: #10882

Previous pr has a problem because it was only adding proper label on defaulter. Not all resources are updated by defaulter, and not all resources are created using store, some like Dataplane are created and updated in controllers. These need to add labels explicitly.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by jakubdyszkiewicz** [comment]
so what's different between this and the previous PR?

**Comment by lobkovilya** [comment]
So basically we have 3 ways resources are created on k8s:
* store.go
* directly applied using kubectl (we catch that in webhook)
* controllers

It would be ideal all these places to use the same code that sets labels. But if it requires a lot of code changes we at least should try to have all controller to share the code

File: pkg/core/resources/model/resource.go [hunk 1]
```diff
@@ -461,7 +468,14 @@ func ComputeLabels(r Resource, mode config_core.CpMode, isK8s bool, systemNamesp
 		}
 	}
 
-	if ns, ok := labels[mesh_proto.KubeNamespaceTag]; ok && r.Descriptor().IsPolicy && r.Descriptor().IsPluginOriginated {
```
Comment (Automaat) [inline pkg/core/resources/model/resource.go:464]: we should refactor computeLabals, to be more specific and probably use some form of builder

**Comment by lobkovilya** [comment]
Resources such as Dataplane, MeshGatewayInstance and etc should be added to the `defaulter` webhook. In this case the `ComputeLabels` method will be called automatically for them.

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**